R/extract.R

```diff
@@ -98,8 +98,25 @@ filter_chunk_end_positions <- function(starts, ends) {
   code_ends
 }
 
+bookdown_engines <- c(
+  # full name    # supported abbreviation
+  "theorem",     "thm",
+  "lemma",       "lem",
+  "corollary",   "cor",
+  "proposition", "prp",
+  "conjecture",  "cnj",
+  "definition",  "def",
+  "example",     "exm",
+  "exercise",    "exr",
+  "hypothesis",  "hyp"
+)
+
 defines_knitr_engine <- function(start_lines) {
-  engines <- names(knitr::knit_engines$get())
+  engines <- c(
+    names(knitr::knit_engines$get()), # knitr
+    bookdown_engines,                 # bookdown
```

Comment:
Not directly related to the current issue, but still important to be included here for #797